marklogic-data-hub/src/test/java/com/marklogic/hub/HubTestBase.java

```diff
@@ -964,6 +987,24 @@ protected void installHubArtifacts(HubConfig hubConfig, boolean force) {
         }
     }
 
+    protected void generateFunctionMetadata(HubConfig hubConfig) {
+        logger.debug("Generating function metadata into MarkLogic");
+        List<Command> commands = new ArrayList<>();
+
+        commands.add(generateFunctionMetadataCommand);
+
+        SimpleAppDeployer deployer = new SimpleAppDeployer(((HubConfigImpl)hubConfig).getManageClient(), ((HubConfigImpl)hubConfig).getAdminManager());
+        deployer.setCommands(commands);
+        deployer.deploy(hubConfig.getAppConfig());
+
+        // Provides some time for post-commit triggers to complete
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
```

Comment:
I have a custom function test that is failing because the post commit trigger is taking longer than 3 sec to complete. It was fine locally, I tested it 5 times in a row. @rjrudin @srinathgit @aebadirad @ryanjdew, are you guys OK if I bump it up to 5sec?

Reply:
Yup, fine with me. Though I think a nicer approach would be to check every second to see if the schemas database now has the expected artifact in it.